examples/azure_ml_orchestrator_flavor/orchestrator/azure_ml_pipelines_orchestrator.py

```diff
@@ -73,356 +69,337 @@ def settings_class(self) -> Optional[Type["BaseSettings"]]:
-        fileio.makedirs(self.pipeline_directory)
-        pipeline_file_path = os.path.join(
-            self.pipeline_directory, f"{orchestrator_run_name}.yaml"
-        )
-
-        # Set the run name, which AzureMLPipelines reads from this attribute of the
-        # pipeline function
-        setattr(
-            _construct_kfp_pipeline,
-            "_component_human_name",
-            orchestrator_run_name,
-        )
-        pipeline_config = AzureMLPipelinesPipelineConf()
-        pipeline_config.add_pipeline_label(
-            "pipelines.kubeflow.org/cache_enabled", "false"
-        )
-        AzureMLPipelinesCompiler().compile(
-            _construct_kfp_pipeline,
-            pipeline_file_path,
-            azuremlpipelines_pipeline_conf=pipeline_config,
-        )
-        logger.info(
-            "Writing AzureMLPipelines workflow definition to `%s`.", pipeline_file_path
-        )
+    def prepare_or_run_pipeline(
+            self, deployment: "PipelineDeployment", stack: "Stack"
+    ) -> None:
+        """Prepares or runs a pipeline on AzureML.
 
+        Args    :
+            deployment: The deployment to prepare or run.
+            stack: The stack to run on.
+        """
         if deployment.schedule:
             logger.warning(
-                "The AzureMLPipelines Orchestrator currently does not support the "
+                "The AzureML Orchestrator currently does not support the "
                 "use of schedules. The `schedule` will be ignored "
                 "and the pipeline will be run immediately."
             )
 
-        kubernetes_context = self.config.kubernetes_context
-        if kubernetes_context:
-            logger.info(
-                "Running AzureMLPipelines pipeline in kubernetes context '%s' and "
-                "namespace '%s'.",
-                kubernetes_context,
-                self.config.kubernetes_namespace,
+        credential = AzureCliCredential()
+        ml_client = MLClient.from_config(credential=credential)
+        # get the compute target to use to run our pipeline
+        # can be single node or cluster
+        cpu_compute_target = "cpu-cluster"
+        # cpu_cluster = ml_client.compute.get(cpu_compute_target)
+
+        # get/create the environment that will be used to run each step of the pipeline
+        # each step can have its own environment, or they can all use the same one
+        # Here you specify your dependencies etc.
+        # This can be a Docker image.
+        custom_env_name = "zenml-test-env"
+        source_directory = get_source_root()
+        orchestrator_run_name = get_orchestrator_run_name(
+            pipeline_name=deployment.pipeline.name
+        ).replace("_", "-")
+        docker_image_builder = PipelineDockerImageBuilder()
+        image_name_or_digest,_,_ = docker_image_builder.build_docker_image(docker_settings=deployment.pipeline_configuration.docker_settings,
```

Comment:
Instead of having to manually build the image here, your orchestrator already inherits from `ContainerizedOrchestrator` which takes care of building this image. To get the image digest, you can simply call `self.get_image(...)`. See the other orchestrators for reference: https://github.com/zenml-io/zenml/blob/58cfdbd857cde809aa32f0bc24964101a1ac0175/src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py#L539C1-L541